enable/brush.py

```diff
@@ -0,0 +1,224 @@
+# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Brush classes to make filling regions cleaner. """
+
+from abc import abstractmethod
+from operator import attrgetter
+
+from numpy import array
+
+from traits.api import (
+    ABCHasStrictTraits, Enum, Event, Float, HasStrictTraits, Instance, List,
+    Range, Tuple, observe
+)
+
+from enable.colors import transparent_color_trait
+
+
+class Brush(ABCHasStrictTraits):
+    """ A brush describes how to fill the pixels of a region.
+
+    To use a brush when drawing, call set_brush with the current gc as the
+    argument.
+    """
+
+    #: A trait which fires when the brush is updated.
+    updated = Event()
+
+    @abstractmethod
+    def set_brush(self, gc):
+        """ Apply the brush settings to a graphics context.
+
+        Parameters
+        ----------
+        gc : graphics context
+            The graphics context to use the brush with.
+        """
+        raise NotImplementedError()
+
+    @observe("+update")
+    def observe_update_traits(self, event):
+        self.updated = True
+
+
+class ColorBrush(Brush):
+    """ A simple brush that paints a solid color. """
+
+    #: The color to brush the region with.
+    color = transparent_color_trait(update=True)
```

Comment:
I feel like `ColorTrait("transparent", update=True)` would read better here. `transparent_color_trait` seems like something that can't have its color reassigned.